src/common/collection.h

```diff
@@ -204,6 +204,9 @@ void dt_collection_hint_message(const dt_collection_t *collection);
 /** returns the image offset in the collection */
 int dt_collection_image_offset(int imgid);
 
+/** returns the image offset for the specified collection */
+int dt_collection_image_offset_with_collection(const dt_collection_t *collection, int imgid);
```

Comment:
Since you only use that function internally in `collection.c` I would prefer to make it `static` and not expose it in the header file.